app/views/assignments/_form.html.erb

```diff
@@ -215,10 +216,30 @@
                              form: f } %>
       <% end %>
     </div>
-
      <p id='assignment_file_add_link'>
         <%= add_assignment_file_link t('assignment.add_required_file'), f %>
      </p>
+     <br>
+     <% if @show_checkbox_required_files == true %>
+	   <%= f.check_box :only_required_files,
+	                      checked: @assignment.only_required_files,
+	                      onclick: 'set_onbeforeunload(true);' 
+	   %>
+	   <%= f.label :only_required_files,
+	                 t('required_file_warning'),
+	                 { id: "assignment_only_required_files_label" }
+	   %>
+	 <% else %>
+	   <%= f.check_box :only_required_files,
```

Comment:
It is going to be much easier. But then when we want to create a new file the following sequence will appear:

Add filename(s) that students would be required to submit:
Add a Required File (a link)
(checkbox) Accept no other files besides the listed required files. This also restricts the file extensions.

which, I think, is not the most graceful way, as the appearance checkbox does not make sense until we see the form generated by clicking on "Add a Required File".